woocommerce/woocommerce
# Redirect to WooPayments Connect page in the Payments task

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44499 



### How to test the changes in this Pull Request:

This change redirects a user to WooPayments Connect page in the Payments task if the merchant is from a supported country and WooPayments is installed.



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Scenario 1: WooPayments is installed and **active**. Merchant country is **supported** by WooPayments
1. Go to WooCommerce => Settings and choose a supported country (US for example)
2. Go back to WooCommerce => Home and click on the 3rd task: Get paid with WooPayments

![Screenshot 2024-03-14 at 12 29 17](https://github.com/woocommerce/woocommerce/assets/12542046/2de48b06-131b-40c5-833b-e56e4b5b2cd3)

3. You should be redirected to the Payments > Connect page
<img width="1563" alt="Screenshot 2024-03-13 at 09 34 08" src="https://github.com/woocommerce/woocommerce/assets/12542046/675d08f7-c243-4bca-929a-169d155cb56e">

🍵 

Scenario 2: WooPayments is installed and **active**. Merchant country is **unsupported** by WooPayments
1. Go to WooCommerce => Settings and choose a supported country (Ukraine for example)

2. Go back to WooCommerce => Home and click on the 3rd task: Get paid

![Screenshot 2024-03-14 at 12 30 24](https://github.com/woocommerce/woocommerce/assets/12542046/0895003a-9e71-4c6d-a315-179b9d12b276)

3. You should be redirected to the Task details
<img width="1563" alt="Screenshot 2024-03-13 at 09 33 18" src="https://github.com/woocommerce/woocommerce/assets/12542046/9c8459a8-cefa-4c78-8e2c-0590cd23df39">

🍵

Scenario 3: WooPayments is **partial onboarded state**

1. Go to WooCommerce => Settings and choose a supported country (US for example)
2. Go back to WooCommerce => Home and click on the 3rd task: Get paid with WooPayments
3. Get redirected to the connect page, proceed
4. Start new onboarding in the sandbox mode, leave at the Stripe KYC step immediately
5. Go back to WooCommerce => Home and click on the 3rd task: Get paid with WooPayments
6. You should be redirected to WooPayments -> Overview page

![image](https://github.com/woocommerce/woocommerce/assets/79862886/76ca1d1e-9071-4e87-8901-84330355dc46)

🍵 

Scenario 4: WooPayments is **not installed**
You should get redirected to `wp-admin/admin.php?page=wc-admin&path=%2Fwc-pay-welcome-page&from=WCADMIN_PAYMENT_TASK`

![image](https://github.com/woocommerce/woocommerce/assets/79862886/55df7968-df1a-4432-a24e-11049bc3e65e)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
